.github/workflows/ci.yaml

```diff
@@ -0,0 +1,157 @@
+name: Build, Test and Publish
+
+on:
+  push:
+    tags: [ '**' ]
+    branches: [ '**' ]
+
+env:
+  DOCKER_REGISTRY: bakdata
+  SENTENCE_PRODUCER_IMAGE_NAME: kpops-demo-sentence-producer
+  WORD_COUNT_APPLICATION_IMAGE_NAME: kpops-demo-word-count-app
+  WORKING_DIRECTORY_APP: "./word-count"
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Build
+        uses: bakdata/ci-templates/actions/java-gradle-build@v1.7.0
+        with:
+          java-distribution: "microsoft"
+          java-version: "17"
+          gradle-version: "wrapper"
+          working-directory: ${{env.WORKING_DIRECTORY_APP}}
+
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
+    needs: build
+
+    steps:
+      - name: Test
+        uses: bakdata/ci-templates/actions/java-gradle-test@v1.7.0
+        with:
+          java-distribution: "microsoft"
+          java-version: "17"
+          gradle-version: "wrapper"
+          working-directory: ${{env.WORKING_DIRECTORY_APP}}
+
+  build-jib:
+      name: Build tarball images
+      runs-on: ubuntu-22.04
+      needs: test
+
+      steps:
+        - name: Check out repository
+          uses: bakdata/ci-templates/actions/checkout@1.32.0
+
+        - name: Set up Gradle
+          uses: bakdata/ci-templates/actions/java-gradle-setup@v1.16.0
+          with:
+            java-distribution: "microsoft"
+            java-version: "17"
+            gradle-version: "wrapper"
+
+        - name: Build Docker images
+          run: |
+            ./gradlew jibBuildTar \
+             --info --stacktrace \
+             --image=$SENTENCE_PRODUCER_IMAGE_NAME:${{ github.run_id }} \
+             -Djib.outputPaths.tar=build/$SENTENCE_PRODUCER_IMAGE_NAME.tar -Djib.container.mainClass=com.bakdata.kpops.examples.SentenceProducer
+
+            ./gradlew jibBuildTar \
+             --info --stacktrace \
+             --image=$WORD_COUNT_APPLICATION_IMAGE_NAME:${{ github.run_id }} \
+             -Djib.outputPaths.tar=build/$WORD_COUNT_APPLICATION_IMAGE_NAME.tar -Djib.container.mainClass=com.bakdata.kpops.examples.WordCountApplication
+          shell: bash
+          working-directory: ${{env.WORKING_DIRECTORY_APP}}
+
+        - name: Upload ${{env.SENTENCE_PRODUCER_IMAGE_NAME}} image artifact
```

Comment:
Please avoid using artifacts as it is slow. Just have another step next that directly pushes the images